pilot/pkg/xds/endpoint_builder.go

```diff
@@ -78,9 +82,18 @@ func (b EndpointBuilder) Key() string {
 	if b.service != nil {
 		params = append(params, string(b.service.Hostname)+"/"+b.service.Attributes.Namespace)
 	}
+	if b.networkView != nil {
+		for nw := range b.networkView {
+			params = append(params, nw)
+		}
```

Comment:
this is not right.. the order of iteration on a map is non-deterministic. So the key string will change from one call ot another for the same endpoint.